jpype/_core.py

```diff
@@ -215,9 +215,19 @@ def startJVM(*args, **kwargs):
                         % (','.join([str(i) for i in kwargs])))
 
     try:
+        import locale
+        # Gather a list of locale settings that Java may override (excluding LC_ALL)
+        categories = [getattr(locale, i) for i in locale.__all__ if i.startswith('LC_') and i != 'LC_ALL']
```

Comment:
Accessing ``__all__`` is fragile (https://docs.python.org/3/tutorial/modules.html?highlight=__all__#importing-from-a-package), even if in practice ``locale`` always has it.

Rather to use ``dir`` to access the attributes of ``locale``:

```
>>> print(', '.join(sorted(name for name in locale.__all__ if name.startswith('LC_'))))
LC_ALL, LC_COLLATE, LC_CTYPE, LC_MESSAGES, LC_MONETARY, LC_NUMERIC, LC_TIME

>>> print(', '.join(name for name in sorted(dir(locale)) if name.startswith('LC_')))
LC_ALL, LC_COLLATE, LC_CTYPE, LC_MESSAGES, LC_MONETARY, LC_NUMERIC, LC_TIME
```

Or, if you want to get the attribute values, we can use ``vars``:

```
categories = [v for k, v in vars(locale).items() if k.startswith('LC_')]
```